src/main/java/uk/gov/hmcts/reform/professionalapi/controller/external/ProfessionalExternalUserController.java

```diff
@@ -100,4 +107,44 @@
 
         return profUsersEntityResponse;
     }
+
+    @ApiOperation(
+            value = "Modify roles for user",
+            authorizations = {
+                    @Authorization(value = "ServiceAuthorization"),
+                    @Authorization(value = "Authorization")
+            }
+    )
+    @ApiResponses({
+            @ApiResponse(
+                    code = 201,
+                    message = "User Roles has been added",
+                    response = OrganisationResponse.class
+            ),
+            @ApiResponse(
+                    code = 403,
+                    message = "Forbidden Error: Access denied"
+            ),
+            @ApiResponse(
+                    code = 404,
+                    message = "Not Found"
+            )
+    })
+    @PutMapping(
+            path = "/users/{userId}",
+            produces = MediaType.APPLICATION_JSON_UTF8_VALUE
+    )
+    @ResponseBody
+    @Secured("pui-user-manager")
+    public ResponseEntity<ModifyUserRolesResponse> modifyRolesForExistingUserOfExternalOrganisation(
+            @RequestBody ModifyUserProfileData modifyUserProfileData,
+            @OrgId String orgId,
+            @PathVariable("userId") String userId
+    ) {
+
+        log.info("Received request to update user roles of an organisation...");
+        profExtUsrReqValidator.validateModifyRolesRequest(modifyUserProfileData, userId);
```

Comment:
can validation be done in superController ? because Internal and external can use same validation then.